### PR TITLE
Global action, tail log files, and clear log files

### PR DESCRIPTION
**Pull request by fillmorejd**
Adding Global action - can disable with config file
Adding tail actions for process Stdout and Stderr
Adding clear log files for global, instance, and processes

### CODE REVIEW

File: app/src/Controller.php [hunk 1]
```diff
@@ -51,12 +51,122 @@ public function index(Request $request, Response $response, array $args)
         in_array($layout, ['grid', 'list']) or $layout = 'grid';
 
         $template = $this->app['twig']->loadTemplate(sprintf('layout/%s.twig', $layout));
-        $response->setContent($template->render(['instances' => $this->app['manager']->getAll()]));
+        $response->setContent($template->render(['instances' => $this->app['manager']->getAll(),'baseUrl'=> $this->app->getConfig('dir','/'), 'globalActions' => $this->app->getConfig('globalActions',false)]));
```
Comment (sagikazarmark) [inline app/src/Controller.php]: This is a bit complicated. Can you please expand it to multi lines?

File: app/views/template.twig [hunk 2]
```diff
@@ -56,5 +71,9 @@
                 $('[data-toggle="tooltip"]').tooltip()
             })
         </script>
+        {% block script %}{% endblock %}
     </body>
-</html>
+{% if doNotClose is defined and doNotClose == true %}
```
Comment (sagikazarmark) [inline app/views/template.twig]: If you invert the condition, you can remove the else part.

File: config/config.yml [hunk 3]
```diff
@@ -1 +1,2 @@
 waitForSupervisor: true
+globalActions: true
```
Comment (sagikazarmark) [inline config/config.yml]: A new line is missing from here.

File: app/src/Controller.php [hunk 4]
```diff
@@ -51,12 +51,122 @@ public function index(Request $request, Response $response, array $args)
...
+     * @param Request  $request
+     * @param Response $response
+     * @param array    $args
+     *
+     * @return Response
+     */
+    public function restartAllServers(Request $request, Response $response, array $args)
+    {
+        $instances = $this->app['manager']->getAll();
+        foreach ($instances as $instance) {
```
Comment (sagikazarmark) [inline app/src/Controller.php:97]: Can you please add a few newlines to increase readability? Like before and after loops?

File: app/src/Controller.php [hunk 5]
```diff
@@ -165,4 +275,102 @@ public function stopProcess(Request $request, Response $response, array $args)
...
+             $this->getLog($response, $instance, $args['process'],'Stderr');
+         } catch (Exception $ex) {
+             $response->setContent($ex->getMessage());
+         }
+    }
+    
+    /**
+     * Tail -f for stdout or stderr for a process
+     * 
+     * @param type $response
```
Comment (sagikazarmark) [inline app/src/Controller.php]: These "types" are incorrect I think.

File: app/src/Controller.php [hunk 6]
```diff
@@ -165,4 +275,102 @@ public function stopProcess(Request $request, Response $response, array $args)
...
+        while (true) 
+        {
+            // Check if client is still connected. 
+            if (connection_aborted()) 
+            {
+                return;
+            }
+            usleep(300000); //0.3 s
+            //Get log until no more from last offset. 
+            $bufferOverFlow = true;
```
Comment (sagikazarmark) [inline app/src/Controller.php:361]: Some newlines in this function could also increase readability.

File: app/views/actions_all.twig [hunk 7]
```diff
@@ -1,3 +1,4 @@
 <a href="start/{{ name }}" title="Start all" data-toggle="tooltip"><span class="glyphicon glyphicon-play text-success"></span></a>
 <a href="restart/{{ name }}" title="Restart all" data-toggle="tooltip"><span class="glyphicon glyphicon-refresh"></span></a>
 <a href="stop/{{ name }}" title="Stop all" data-toggle="tooltip"><span class="glyphicon glyphicon-stop text-danger"></span></a>
+<a href="clearAllProcessLogs/{{ name }}" title="Clear log files" data-toggle="tooltip"><span class="glyphicon glyphicon-remove"></span></a>
```
Comment (sagikazarmark) [inline app/views/actions_all.twig]: A new line is missing from here.

**Comment by sagikazarmark** [comment]
This is quite awesome @fillmorejd, thank you very much again. I added some comments. Can you please also fix CS issues highlighted by StyleCI. PHP CS Fixer rules are available in the project, but please see the .php_cs file for details.

I also would like to ask you to squash your commits and rebase the branch from master to have a continuous history.

**Comment by fillmorejd** [comment]
Thanks @sagikazarmark 

I will work on the changes.

**Comment by fillmorejd** [comment]
Sorry for the delay on this.  Work has gotten really crazy.  Hopefully I will get back to this later this week.

**Comment by sagikazarmark** [comment]
:+1:

**Comment by fillmorejd** [comment]
All checks are now passing

**Comment by sagikazarmark** [comment]
LGTM, thanks. Can you please squash your commits?

**Comment by fillmorejd** [comment]
Only 4 months later, but I finally took the time to figure out how to squash commit correctly.  Hopefully I did do it correctly.

**Pull request merged**